File: backend/queues/src/main/java/com/griddynamics/msd365fp/manualreview/queues/model/dto/DFPItemDTO.java
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

package com.griddynamics.msd365fp.manualreview.queues.model.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class DFPItemDTO {
    private String purchaseId;
    private OffsetDateTime merchantLocalDate;
    private BigDecimal totalAmount;
    private BigDecimal totalAmountInUSD;
    private BigDecimal salesTax;
    private BigDecimal salesTaxInUSD;
    private String currency;
    private Integer riskScore;
    private String merchantRuleDecision;
    private String reasonCodes;
    private LAUser user;
    private LADeviceContext deviceContext;
    private Boolean userRestricted;

    @NoArgsConstructor
    @AllArgsConstructor
    @Data
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class LAUser {
        private String email;
        private String userId;
    }

    @NoArgsConstructor
    @AllArgsConstructor
    @Data
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class LADeviceContext {
        private String ipAdress;
    }
}

File: frontend/src/data-services/api-services/models/item/dfp-item-dto.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { DeviceContextDTO } from '../device-context-dto';

export interface DfpItemDto {
    purchaseId: string;
    merchantLocalDate: string;
    totalAmount: number;
    totalAmountInUSD: number;
    salesTax: number;
    salesTaxInUSD: number;
    currency: string;
    riskScore: number;
    merchantRuleDecision: string;
    reasonCodes: string;
    user: {
        email:string,
        userIs: string;
    };
    deviceContext: DeviceContextDTO;
    userRestricted: boolean;
}
